gef-ui/server/utils/deal-helpers.test.js

```diff
@@ -103,6 +104,58 @@ describe('makerCanReSubmit', () => {
   });
 });
 
+describe('canAmendFacilities', () => {
+  const MOCK_BANK = { id: '123' };
+
+  it('Should return TRUE for a maker and acknowledged deal at the same bank', () => {
+    const mockDeal = MOCK_BASIC_DEAL;
+    mockDeal.status = CONSTANTS.DEAL_STATUS.UKEF_ACKNOWLEDGED;
+    mockDeal.bank = MOCK_BANK;
+
+    const mockUserSession = MOCK_REQUEST;
+    mockUserSession.roles = [CONSTANTS.USER_ROLES.MAKER];
+    mockUserSession.bank = MOCK_BANK;
+
+    expect(canAmendFacilities(mockUserSession, mockDeal)).toEqual(true);
+  });
+
+  it('Should return FALSE if deal is not acknowleged', () => {
+    const mockDeal = MOCK_BASIC_DEAL;
+    mockDeal.status = CONSTANTS.DEAL_STATUS.SUBMITTED_TO_UKEF;
+    mockDeal.bank = MOCK_BANK;
+
+    const mockUserSession = MOCK_REQUEST;
+    mockUserSession.roles = [CONSTANTS.USER_ROLES.MAKER];
+    mockUserSession.bank = MOCK_BANK;
+
+    expect(canAmendFacilities(mockUserSession, mockDeal)).toEqual(false);
+  });
+
+  it('Should return FALSE if user is not a maker', () => {
+    const mockDeal = MOCK_BASIC_DEAL;
+    mockDeal.status = CONSTANTS.DEAL_STATUS.UKEF_ACKNOWLEDGED;
+    mockDeal.bank = MOCK_BANK;
+
+    const mockUserSession = MOCK_REQUEST;
+    mockUserSession.roles = [CONSTANTS.USER_ROLES.CHECKER];
+    mockUserSession.bank = MOCK_BANK;
+
+    expect(canAmendFacilities(mockUserSession, mockDeal)).toEqual(false);
+  });
+
+  it('Should return FALSE if user is a maker for a different bank to the deal', () => {
+    const mockDeal = MOCK_BASIC_DEAL;
```

Comment:
Can we add an assertion where the user session does not have `bank` property please.